File: 2024_java_basics/09-for-loops/tasks/09-left-and-right-sum.java
import java.util.Scanner;

public class LeftAndRightSum {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int rows = scanner.nextInt();
        int leftSum = 0;
        int rightSum = 0;

        // Read and process 2 * rows numbers
        for (int i = 0; i < 2 * rows; i++) {
            int number = scanner.nextInt();
            if (i < rows) {
                leftSum += number;
            } else {
                rightSum += number;
            }
        }

        // Compare sums and print results
        if (leftSum == rightSum) {
            System.out.println("Yes, sum = " + leftSum);
        } else {
            System.out.println("No, diff = " + Math.abs(leftSum - rightSum));
        }
        
        scanner.close();
    }
}

File: 2022_c#_basics/09-for-loop/09-left-and-right-sum.cs
using System;

namespace _09_Left_and_Right_Sum
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int rows = int.Parse(Console.ReadLine());
            int leftSum = 0;
            int rightSum = 0;

            for (int i = 0; i < 2 * rows; i++)
            {
                int number = int.Parse(Console.ReadLine());
                if (i < rows)
                {
                    leftSum += number;
                }
                else
                {
                    rightSum += number;
                }
            }

            if (leftSum == rightSum)
            {
                Console.WriteLine($"Yes, sum = {leftSum}");
            }
            else
            {
                Console.WriteLine($"No, diff = {Math.Abs(leftSum - rightSum)}");
            }
        }
    }
}
